go/x509util/x509util.go

```diff
@@ -0,0 +1,576 @@
+		commaAppend(&buf, "Digital Signature")
+	}
+	if k&x509.KeyUsageContentCommitment != 0 {
+		commaAppend(&buf, "Content Commitment")
+	}
+	if k&x509.KeyUsageKeyEncipherment != 0 {
+		commaAppend(&buf, "Key Encipherment")
+	}
+	if k&x509.KeyUsageDataEncipherment != 0 {
+		commaAppend(&buf, "Data Encipherment")
+	}
+	if k&x509.KeyUsageKeyAgreement != 0 {
+		commaAppend(&buf, "Key Agreement")
+	}
+	if k&x509.KeyUsageCertSign != 0 {
+		commaAppend(&buf, "Certificate Signing")
+	}
+	if k&x509.KeyUsageCRLSign != 0 {
+		commaAppend(&buf, "CRL Signing")
+	}
+	if k&x509.KeyUsageEncipherOnly != 0 {
+		commaAppend(&buf, "Encipher Only")
+	}
+	if k&x509.KeyUsageDecipherOnly != 0 {
+		commaAppend(&buf, "Decipher Only")
+	}
+	return buf.String()
+}
+
+func extKeyUsageToString(u x509.ExtKeyUsage) string {
+	switch u {
+	case x509.ExtKeyUsageAny:
+		return "Any"
+	case x509.ExtKeyUsageServerAuth:
+		return "TLS Web server authentication"
+	case x509.ExtKeyUsageClientAuth:
+		return "TLS Web client authentication"
+	case x509.ExtKeyUsageCodeSigning:
+		return "Signing of executable code"
+	case x509.ExtKeyUsageEmailProtection:
+		return "Email protection"
+	case x509.ExtKeyUsageIPSECEndSystem:
+		return "IPSEC end system"
+	case x509.ExtKeyUsageIPSECTunnel:
+		return "IPSEC tunnel"
+	case x509.ExtKeyUsageIPSECUser:
+		return "IPSEC user"
+	case x509.ExtKeyUsageTimeStamping:
+		return "Time stamping"
+	case x509.ExtKeyUsageOCSPSigning:
+		return "OCSP signing"
+	case x509.ExtKeyUsageMicrosoftServerGatedCrypto:
+		return "Microsoft server gated cryptography"
+	case x509.ExtKeyUsageNetscapeServerGatedCrypto:
+		return "Netscape server gated cryptography"
+	default:
+		return "Unknown"
+	}
+}
+
+func attributeOidToString(oid asn1.ObjectIdentifier) string {
+	switch {
+	case oid.Equal(OidCountry):
+		return "Country"
+	case oid.Equal(OidOrganization):
+		return "Organization"
+	case oid.Equal(OidOrganizationalUnit):
+		return "OrganizationalUnit"
+	case oid.Equal(OidCommonName):
+		return "CommonName"
+	case oid.Equal(OidSerialNumber):
+		return "SerialNumber"
+	case oid.Equal(OidLocality):
+		return "Locality"
+	case oid.Equal(OidProvince):
+		return "Province"
+	case oid.Equal(OidStreetAddress):
+		return "StreetAddress"
+	case oid.Equal(OidPostalCode):
+		return "PostalCode"
+	case oid.Equal(OidPseudonym):
+		return "Pseudonym"
+	case oid.Equal(OidTitle):
+		return "Title"
+	case oid.Equal(OidDnQualifier):
+		return "DnQualifier"
+	case oid.Equal(OidName):
+		return "Name"
+	case oid.Equal(OidSurname):
+		return "Surname"
+	case oid.Equal(OidGivenName):
+		return "GivenName"
+	case oid.Equal(OidInitials):
+		return "Initials"
+	case oid.Equal(OidGenerationQualifier):
+		return "GenerationQualifier"
+	default:
+		return oid.String()
+	}
+}
+
+// NameToString creates a string description of a pkix.Name object.
+func NameToString(name pkix.Name) string {
+	var result bytes.Buffer
+	addSingle := func(item string, prefix string) {
```

Comment:
Having looked further down at the usage of this function and addList, it would make a lot of the places where you use it more readable if the two parameters were passed in the other way around, i.e `addSingle(prefix, item)`, as that is the order they're actually added to the string.  There was a moment where I was like 'wait, so the string is going to end with an '='??  This would fix that.  Just for ease of understanding the code :)